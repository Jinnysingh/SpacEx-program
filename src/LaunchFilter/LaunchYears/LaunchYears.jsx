import React from "react";

function LaunchYear(props) {
  return (
      <div className="launch-sec">
      <div>
        <u>Launch Year</u>
      </div>
      {props.launchFilters.map((items) => (
        <button type="button" className="btn btn-success text-dark launch-btn active-btn" onClick={(event) => props.handleFilters(event, "Launch Years")}>
          {items}
        </button>
      ))}
    </div>
  );
}

export default LaunchYear;
