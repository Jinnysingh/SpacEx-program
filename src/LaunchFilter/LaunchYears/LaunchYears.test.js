import {mount} from "enzyme"
import toJSON from "enzyme-to-json"
import React from "react"
import LaunchYears from "./LaunchYears.jsx"

let launchFilters = [2006, 2007, 2008, 2009, 2010, 2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018, 2019, 2020]

describe('<launchYears />', () => {
    const wrapper = mount(
      <LaunchYears launchFilters={launchFilters} />
    );

test("testing the filters", ()=>{
  expect(launchFilters).not.toBe(0);
  expect(toJSON(wrapper)).toMatchSnapshot();
})
})