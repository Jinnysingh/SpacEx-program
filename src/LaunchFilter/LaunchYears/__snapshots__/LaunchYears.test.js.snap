// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<launchYears /> testing the filters 1`] = `
<LaunchYear
  launchFilters={
    Array [
      2006,
      2007,
      2008,
      2009,
      2010,
      2011,
      2012,
      2013,
      2014,
      2015,
      2016,
      2017,
      2018,
      2019,
      2020,
    ]
  }
>
  <div
    className="launch-sec"
  >
    <div>
      <u>
        Launch Year
      </u>
    </div>
    <button
      className="btn btn-success text-dark launch-btn active-btn"
      onClick={[Function]}
      type="button"
    >
      2006
    </button>
    <button
      className="btn btn-success text-dark launch-btn active-btn"
      onClick={[Function]}
      type="button"
    >
      2007
    </button>
    <button
      className="btn btn-success text-dark launch-btn active-btn"
      onClick={[Function]}
      type="button"
    >
      2008
    </button>
    <button
      className="btn btn-success text-dark launch-btn active-btn"
      onClick={[Function]}
      type="button"
    >
      2009
    </button>
    <button
      className="btn btn-success text-dark launch-btn active-btn"
      onClick={[Function]}
      type="button"
    >
      2010
    </button>
    <button
      className="btn btn-success text-dark launch-btn active-btn"
      onClick={[Function]}
      type="button"
    >
      2011
    </button>
    <button
      className="btn btn-success text-dark launch-btn active-btn"
      onClick={[Function]}
      type="button"
    >
      2012
    </button>
    <button
      className="btn btn-success text-dark launch-btn active-btn"
      onClick={[Function]}
      type="button"
    >
      2013
    </button>
    <button
      className="btn btn-success text-dark launch-btn active-btn"
      onClick={[Function]}
      type="button"
    >
      2014
    </button>
    <button
      className="btn btn-success text-dark launch-btn active-btn"
      onClick={[Function]}
      type="button"
    >
      2015
    </button>
    <button
      className="btn btn-success text-dark launch-btn active-btn"
      onClick={[Function]}
      type="button"
    >
      2016
    </button>
    <button
      className="btn btn-success text-dark launch-btn active-btn"
      onClick={[Function]}
      type="button"
    >
      2017
    </button>
    <button
      className="btn btn-success text-dark launch-btn active-btn"
      onClick={[Function]}
      type="button"
    >
      2018
    </button>
    <button
      className="btn btn-success text-dark launch-btn active-btn"
      onClick={[Function]}
      type="button"
    >
      2019
    </button>
    <button
      className="btn btn-success text-dark launch-btn active-btn"
      onClick={[Function]}
      type="button"
    >
      2020
    </button>
  </div>
</LaunchYear>
`;
